api/app/auto_spatial_advisory/critical_hours.py

```diff
@@ -0,0 +1,457 @@
+        advisory_fuel_stat = row[0]
+        sfms_fuel_type = row[1]
+        key = sfms_fuel_type.fuel_type_code
+        if key == "Non-fuel":
+            continue
+        if key in fuel_types_by_area:
+            fuel_types_by_area[key] += advisory_fuel_stat.area
+        else:
+            fuel_types_by_area[key] = advisory_fuel_stat.area
+    return fuel_types_by_area
+
+
+async def get_inputs_for_critical_hours(for_date: date, header: dict, wfwx_stations: List[WFWXWeatherStation]) -> CriticalHoursInputs:
+    """
+    Retrieves the inputs required for computing critical hours based on the station list and for date
+
+    :param for_date: date of interest for looking up dailies and hourlies
+    :param header: auth header for requesting data from WF1
+    :param wfwx_stations: list of stations to compute critical hours for
+    :return: critical hours inputs
+    """
+    unique_station_codes = list(set(station.code for station in wfwx_stations))
+    time_of_interest = get_hour_20_from_date(for_date)
+
+    # get the dailies for all the stations
+    async with ClientSession() as client_session:
+        dailies_by_station_id = await get_dailies_by_station_id(client_session, header, wfwx_stations, time_of_interest)
+        # must retrieve the previous day's observed/forecasted FFMC value from WFWX
+        prev_day = time_of_interest - timedelta(days=1)
+        # get the "daily" data for the station for the previous day
+        yesterday_dailies_by_station_id = await get_dailies_by_station_id(client_session, header, wfwx_stations, prev_day)
+        # get hourly observation history from our API (used for calculating morning diurnal FFMC)
+        hourly_observations_by_station_code = await get_hourly_observations(unique_station_codes, time_of_interest - timedelta(days=4), time_of_interest)
+
+        return CriticalHoursInputs(
+            dailies_by_station_id=dailies_by_station_id,
+            yesterday_dailies_by_station_id=yesterday_dailies_by_station_id,
+            hourly_observations_by_station_code=hourly_observations_by_station_code,
+        )
+
+
+async def calculate_critical_hours_by_zone(db_session: AsyncSession, header: dict, stations_by_zone: Dict[int, List[WFWXWeatherStation]], run_parameters_id: int, for_date: date):
+    """
+    Calculates critical hours for fire zone units by heuristically determining critical hours for each station in the fire zone unit that are under advisory conditions (>4k HFI).
+
+    :param db_session: An async database session.
+    :param header: An authorization header for making requests to WF1.
+    :param stations_by_zone: A dictionary of lists of stations in fire zone units keyed by fire zone unit id.
+    :param run_parameters_id: The RunParameters object (ie. the SFMS run).
+    :param for_date: The date critical hours are being calculated for.
+    :return: A dictionary of critical hours by fuel type per fire zone unit keyed by fire zone unit id.
```

Comment:
Can remove this docstring about what is returned as this function doesn't actually need to return anything.